packages/jest-cli/src/lib/watch_plugin_registry.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {WatchPlugin} from '../types';
+import defaultResolver from '../../../jest-resolve/src/default_resolver';
+
+const RESERVED_KEYS = [
+  0x03, // Jest should handle ctrl-c interrupt
+  0x71, // 'q' is reserved for quit
+];
+
+export default class WatchPluginRegistry {
+  _rootDir: string;
+  _watchPluginsByKey: Map<number, WatchPlugin>;
+
+  constructor(rootDir: string) {
+    this._rootDir = rootDir;
+    this._watchPluginsByKey = new Map();
+  }
+
+  loadPluginPath(pluginModulePath: string) {
+    // $FlowFixMe dynamic require
+    const maybePlugin = require(defaultResolver(pluginModulePath, {
+      basedir: this._rootDir,
+    }));
+
+    // Since we're loading the module from a dynamic path, assert its shape
+    // before assuming it's a valid watch plugin.
+    if (typeof maybePlugin !== 'object') {
```

Comment:
I think we can probably use https://github.com/facebook/jest/tree/master/packages/jest-get-type for all of these